test/fixtures/es-module-require-cache/entry.mjs

```diff
@@ -0,0 +1,4 @@
+import './preload.js';
+import './counter.js';
+import assert from 'assert';
+assert.strictEqual(global.counter, 1);
```

Comment:
Why not just move the .mjs file to test/es-module instead of going through the runner?